### PR TITLE
Upgraded Gradle Wrapper to 8.1.1

### LINKED ISSUE #91: Gradle 8.0 compatibility
When running the tests on Gradle 8.0-rc-2 the tests will fail indicating input capturing doesn't work anymore on Gradle 8.0. This needs to be fixed. See [this build scan](https://scans.gradle.com/s/o5luhevuztr76/tests/:plugin:compatTest8.0-rc-2/de.benediktritter.maven.plugin.development.PluginDescriptorGenerationFuncTest/generates%20a%20plugin%20and%20help%20descriptor%20for%20mojos%20in%20the%20main%20source%20set?top-execution=1) for details.

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- updated gradle wrapper
- supported changes in build.gradle.kts
- updated java8.lock by `:plugin:stutterWriteLocks`
- changed input parameters for tasks, because Gradle 8.x cannot serialize FileCollection and Configuration

It closes #91

### CODE REVIEW

**Comment by britter** [comment]
Hey @nulls, thank you for the PR. Is this something you need included or just build hygene? Asking because I'm considering moving the plugin to https://github.com/gradlex-org and that would include kind of a rewrite anyways.

**Comment by nulls** [comment]
Hi @britter, 
Our project [diktat](https://github.com/saveourtool/diktat) uses this plugin and we are upgrading the Gradle version in the project.
It would be great if it will be included.

Btw, I see that `Sample Project Build` does fail. I guess it's okay till a new version is released, isn't it?

**Comment by britter** [comment]
@nulls okay, I see. I'll have a look into this on the weekend. Unfortunately I think we need to do this differently in order to keep compatibility with released versions.

**Comment by britter** [comment]
@nulls sorry to keep you waiting. This is next on my backlog. I hope to find some time the coming week to merge and release this.

**Review by britter — CHANGES_REQUESTED** [review]
Hey @nulls! Finally got to reviewing this. I think the problem are not the types of the inputs but their annotations. Can you please try if changing them the way I've pointed out makes it work on Gradle 8? That way we can leave the types as they are and don't break binary compat.

File: subprojects/plugin/src/main/kotlin/de/benediktritter/maven/plugin/development/task/GenerateMavenPluginDescriptorTask.kt [hunk 1]
```diff
@@ -43,13 +41,13 @@ import java.io.File
...
     @get:Input
-    abstract val classesDirs: Property<FileCollection>
+    abstract val classesDirs: SetProperty<File>
 
     @get:Internal
     abstract val javaClassesDir: DirectoryProperty
 
     @get:Input
-    abstract val sourcesDirs: Property<FileCollection>
+    abstract val sourcesDirs: SetProperty<File>
```
[the remark this answers is not in the document]
Comment (nulls) [reply subprojects/plugin/src/main/kotlin/de/benediktritter/maven/plugin/development/task/GenerateMavenPluginDescriptorTask.kt]: applied

File: subprojects/plugin/src/main/kotlin/de/benediktritter/maven/plugin/development/task/AbstractMavenPluginDevelopmentTask.kt [hunk 2]
```diff
@@ -27,14 +27,15 @@ import org.gradle.api.artifacts.Configuration
...
+import org.gradle.kotlin.dsl.get
 
 abstract class AbstractMavenPluginDevelopmentTask : DefaultTask() {
 
     @get:Nested
     abstract val pluginDescriptor: Property<MavenPluginDescriptor>
 
     @get:Input
-    abstract val runtimeDependencies: Property<Configuration>
+    abstract val runtimeDependenciesConfigurationName: Property<String>
```
[the remark this answers is not in the document]
Comment (nulls) [reply subprojects/plugin/src/main/kotlin/de/benediktritter/maven/plugin/development/task/AbstractMavenPluginDevelopmentTask.kt]: applied

**Review by britter — APPROVED** [review]
Awesome, thank you!

**Comment by britter** [comment]
@nulls I've released 0.4.2 which contains your fix. Please give it a try!

**Comment by nulls** [comment]
> @nulls I've released 0.4.2 which contains your fix. Please give it a try!

Thanks, now it does work!

**Pull request merged**